### PR TITLE
test_x509: test very long OID

### PR DESCRIPTION
**Pull request by Rezney**
Active Directory creates OIDs long enough to trigger a failure.
This can cause e.g. ipa-server-install failure when installing
with an externally-signed CA.

https://pagure.io/freeipa/issue/7300

### CODE REVIEW

**Comment by Rezney** [comment]
Thanks for checking this. Sure...

File: ipatests/test_ipalib/test_x509.py [hunk 1]
```diff
@@ -151,3 +187,18 @@ def test_load_pkcs7_pem(self):
...
+        Policies OID is longer then 80 chars.
+        """
+        cert = x509.load_pem_x509_certificate(long_oid_cert)
+        ext = cert.extensions.get_extension_for_class(x509crypto.
+                                                      CertificatePolicies)
+
+        assert len(ext.value) == 1
+        assert ext.value[0].policy_identifier.dotted_string == (u'1.3.6.1.4.1.'
+               '311.21.8.8950086.10656446.2706058.12775672.480128.147.'
+               '13466065.13029902')
```
Comment (frasertweedale) [inline ipatests/test_ipalib/test_x509.py]:
```
PEP-8 errors:
./ipatests/test_ipalib/test_x509.py:203:16: E128 continuation line under-indented for visual indent
./ipatests/test_ipalib/test_x509.py:204:16: E128 continuation line under-indented for visual indent
```

You'll also need to rebase on latest master where the tests are fixed.

**Comment by Rezney** [comment]
Should be ok now...

File: ipatests/test_ipalib/test_x509.py [hunk 2]
```diff
@@ -26,6 +26,7 @@
 
 import pytest
 
+from cryptography import x509 as x509crypto
```
Comment (stlaz) [inline ipatests/test_ipalib/test_x509.py]: Pretty pretty please, change the import name to `crypto_x509` so it's consistent with the rest of the code.

Comment (Rezney) [reply ipatests/test_ipalib/test_x509.py]: Makes sense, done...

**Comment by tiran** [comment]
The ticket has no milestone, is push to ``master`` enough?

**Comment by frasertweedale** [comment]
@tiran yeah push to master is fine.

**Comment by tiran** [comment]
master:

* 29d0f8673cf4c944db538244aea7b0dd103979bb test_x509: test very long OID

**Pull request closed without merging**